corehq/apps/translations/app_translations/utils.py

```diff
@@ -202,6 +205,49 @@ def none_or_unicode(val):
     return rows
 
 
+def text_and_audio_changed(current_row, previous_row, headers):
+    text_changed = False
+    audio_path_changed = False
+    for index, header in enumerate(headers):
+        if "default_" in header and previous_row[index] != current_row[header]:
+            text_changed = True
+        if "audio_" in header and previous_row[index] != current_row[header]:
+            audio_path_changed = True
+    return text_changed, audio_path_changed
+
+
+def generate_audio_path(text, lang, previous_path):
+    path_hash = md5(f'{text}_{lang}'.encode()).hexdigest()[:6]
+    file_path_arr = previous_path.split('/')
+    filename_index = len(file_path_arr) - 1
+    complete_filename, extension = os.path.splitext(os.path.basename(previous_path))
+    filename_arr = complete_filename.split('-')
+    filename_arr[len(filename_arr) - 1] = path_hash
+    new_filename = '-'.join(filename_arr) + extension
+    file_path_arr[filename_index] = new_filename
```

Comment:
this logic looks fairly complicated. Can you try to simplify it or add comments/reasoning to help understand it better?

Reply:
Unit tests of this function would also help.

Reply:
It might (not guaranteed ;) ) be more concise to do this using string replacement and regexes instead of splitting and dealing with arrays. `(-\w+)?(\.\w+)$` would capture the old hash (if it exists) and the extension, and then you could replace the hash with the new one.